drivers/sensors/cirque_pinnacle.c

```diff
@@ -192,6 +197,53 @@ void cirque_pinnacle_tune_edge_sensitivity(void) {
     ERA_ReadBytes(0x0168, &temp, 1);
 }
 
+// Perform calibration
+void cirque_pinnacle_calibrate(void) {
+    uint8_t  calconfig;
+    uint16_t timeout_timer;
+
+    // CalConfig1 (Compensation)
+    // Bit 0: Calibrate, 1=calibrate, 0=complete
+    // Bit 1: Background Comp Enable, 1=enable, 0=disable
+    // Bit 2: NERD Comp Enable (No, I don't know what NERD means), 1=enable, 0=disable
+    // Bit 3: Track Error Comp Enable, 1=enable, 0=disable
+    // Bit 4: Tap Comp Enable, 1=enable, 0=disable
+    // Bit 6: Calibration Matrix Disable, 1=disabled, 0=enabled
+    RAP_ReadBytes(CALIBRATION_CONFIG_1, &calconfig, 1);
+    calconfig |= 0x01;
+    RAP_Write(CALIBRATION_CONFIG_1, calconfig);
+
+    // Calibration takes ~100ms according to GT-AN-090624, doubling the timeout just to be safe
+    timeout_timer = timer_read();
+    do {
+        RAP_ReadBytes(CALIBRATION_CONFIG_1, &calconfig, 1);
+    } while ((calconfig & 0x01) && (timer_elapsed(timeout_timer) <= 200));
+
+    cirque_pinnacle_clear_flags();
+}
+
+// Enable/disable cursor smoothing, smoothing is enabled by default
+void cirque_pinnacle_cursor_smoothing(bool enable) {
+    uint8_t feedconfig3;
+
+    // FeedConfig3 (Advanced feature flags)
+    // Bit 0: DualPoint Buttons, 1=enable, 0=disable
+    // Bit 1: Smoothing Disable, 1=disable, 0=enable
+    // Bit 2: Palm/NERD measurements Disable, 1=disable, 0=enable
+    // Bit 3: Noise Avoidance Disable, 1=disable, 0=enable
+    // Bit 4: WRAP lockout Disable
+    // Bit 5: Dynamic EMI adjust Disable
+    // Bit 6: HW EMI detection Disable
+    // Bit 7: SW EMI detection Disable
+    RAP_ReadBytes(FEEDCONFIG_3, &feedconfig3, 1);
```

Comment:
I can do this, but is it really the right choice for a platform independent driver?
> The order of allocation of bit-fields within a unit (high-order to low-order or low-order to high-order) is implementation-defined. The alignment of the addressable storage unit is unspecified.

As an example of a caveat: QMK would have to only ever support little endian microcontrollers (which I believe currently is the case anyway?), otherwise each register would need to have its structure defined once for little endian and then again backwards for big endian.
I think it's safer to import [Cirque's register definition file](https://github.com/cirque-corp/Cirque_Pinnacle_1CA027/blob/master/Additional_Examples/AnyMeas_Example/Pinnacle.h) and use that instead.

Reply:
> I think it's safer to import [Cirque's register definition file](https://github.com/cirque-corp/Cirque_Pinnacle_1CA027/blob/master/Additional_Examples/AnyMeas_Example/Pinnacle.h) and use that instead.

I agree that we should just import the already existing definitions by Cirque because they provide descriptive names (which is the main problem I have with these magic numbers).

> but is it really the right choice for a platform independent driver?

We fully control the build environment which is GCC only at this point (I have a clang patch in the pipeline but not pr'ed yet) and the GCC implementation seems to work for us. Also all current (avr, arm, risc-v) and future platforms (xtensa?) that I can think of are little endian - so that would be fine.

Reply:
I've uploaded an update with a modified version of Cirque's register definition file.
As a start I kept the wordy "HOSTREG__" prefixes to differentiate between regular registers and extended registers, let me know what you think.

Reply:
Thanks! This looks good now.